lib/logflare/sql/dialect_transformer/bigquery.ex

```diff
@@ -0,0 +1,81 @@
+defmodule Logflare.Sql.DialectTransformer.BigQuery do
+  @moduledoc """
+  BigQuery-specific SQL transformations.
+  """
+
+  @behaviour Logflare.Sql.DialectTransformer
+
+  import Logflare.Utils.Guards
+
+  alias Logflare.User
+
+  @impl true
+  def quote_style, do: "`"
+
+  @impl true
+  def dialect, do: "bigquery"
+
+  @impl true
+  def transform_source_name(source_name, %{sources: sources} = data) do
+    source = Enum.find(sources, fn s -> s.name == source_name end)
+
+    token =
+      source.token
+      |> Atom.to_string()
+      |> String.replace("-", "_")
+
+    # byob bq
+    project_id =
+      if is_nil(data.user_project_id), do: data.logflare_project_id, else: data.user_project_id
+
+    # byob bq
+    dataset_id =
+      if is_nil(data.user_dataset_id), do: data.logflare_dataset_id, else: data.user_dataset_id
```

Comment:
what if we did the byob check when building the data instead? and then store only `project_id` and `dataset_id` in the data map?
might be cleaner and we won't "leak" the byob concept into other functions.